### ISSUE TITLE
[bug] CONAN_CMAKE_PROGRAM not respected for CMake Toolchain

### ISSUE THREAD

**Issue by duncanspumpkin** [conan-io/conan#12701]
### Environment Details (include every applicable attribute)
  * Operating System+version:Linux/Windows
  * Compiler+version:gcc/MSVC
  * Conan version:HEAD
  * Python version:3.8

### Steps to reproduce (Include if Applicable)
Have CMake 2.8 installed on your system but also have CMake 3.24 installed to another location.
Specify the CMake version required within CONAN_CMAKE_PROGRAM environment variable.
Try to compile zlib from the conan center recipie. Shouldn't matter the version they have all been converted to use CMake toolchain and specify that CMake minimum version of >2.8.

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.0 or higher is required.  You are running version 2.8.12.2


-- Configuring incomplete, errors occurred!
```

### Source of issue
https://github.com/conan-io/conan/pull/4298 added this feature back in 2019 but when the CMake Toolchain refactor was done in 2020 https://github.com/conan-io/conan/pull/7076 the #4298 change to cmake program were not ported over. 

Problem line is:
https://github.com/conan-io/conan/blame/develop/conan/tools/cmake/cmake.py#L62

Just needs to be updated to follow 4298:
`self._cmake_program = os.getenv("CONAN_CMAKE_PROGRAM") or cmake_program or "cmake"`

I've locally applied this and it fixed my issue.

Of course its completely possible that I am misunderstanding how to set the cmake program path.

**Comment by memsharded** [comment]
Hi @duncanspumpkin 

I'd like to clarify that the environment variables like ``CONAN_CMAKE_PROGRAM`` are not used anymore to configure or define the behavior of the new build system integrations like ``CMakeToolchain``, ``CMakeDeps``, or really anything under the new ``from conan.tools.xxx`` space.

Instead, the possible configuration comes from either environment ``[buildenv]`` in profiles, or configuration ``[conf]`` in profiles.

If the use case is to define the path to a certain CMake version, this can be done in the profile with something like:

```
[buildenv]
PATH=+(path)/my/path/to/cmake-X.Y.Z/bin
```

You can check:
- https://docs.conan.io/en/latest/reference/profiles.html#buildenv
- https://docs.conan.io/en/latest/reference/conanfile/tools/env.html

**Comment by duncanspumpkin** [comment]
Okay thanks after adding the path to the buildenv (And adding `tools.env.virtualenv:auto_use=True` to `[conf]`) it is now working correctly. 

I can't find where I saw CONAN_CMAKE_PROGRAM in the documents anymore now that i look for it (probably found it on stackoverflow). Does https://docs.conan.io/en/latest/reference/build_helpers/cmake.html need to be updated to give some indication that it is sort of legacy? It mentions cmake_program but not CONAN_CMAKE_PROGRAM.

**Comment by memsharded** [comment]
It is documented here: https://docs.conan.io/en/latest/reference/env_vars.html?highlight=conan_cmake_program#conan-cmake-program


The thing about https://docs.conan.io/en/latest/reference/build_helpers/cmake.html, is that such helper might still use it, and it might work. It is the new generators in the new space (https://docs.conan.io/en/latest/reference/conanfile/tools.html), that do not use env-vars for configuration anymore.

It is true that keeping the 1.X docs updated has proven difficult, but we hope that we can release 2.0 GA with a completely new docs from scratch that will be much better, so docs issues will improve.

**Comment by iskunk** [comment]
Finagling `PATH` to get the right CMake executable isn't a good solution, I'm afraid. The desired CMake binary might not be called `cmake` (it could be, for example, `cmake-3.24`), and the bin directory where it resides may contain other binaries that should not be used by the Conan build. This approach is also inconsistent with how other tool paths are specified.

I will submit a PR that adds a tool configuration item to allow setting `_cmake_program`.

**Comment by memsharded** [comment]
#13940 has been merged, for next 2.0.6

However, please take into account that the recommended way to define which CMake is running would be the default environment PATH definition. It has been working robustly for most cases like the ConanCenter CMake package. This new ``conf`` will be recommended only for exceptions like when having an executable with a different name like ``cmake3_15.exe`` or  the like.
